freva-rest/src/freva_rest/tool_api/tool_abstract.py

```diff
@@ -0,0 +1,302 @@
+        default (str, optional): The default value of the parameter. Defaults to None.
+        type (Literal["DataField"]): A description of the parameter's type. Must be "DataField".
+
+    Attributes:
+        facet (str): The databrowser facet used for this parameter. 
+        group (int, optional): The group this search facet belongs to. This can be used to group
+                               different search facets together, for example for comparing
+                               multi model ensembles.
+        multiple(bool, optional): Flag indicating whether multiple values can be selected for the selected facet.
+        predefined_facets(Dict[str, Union[str, List[str]]], optional): A dict containing default values for other search facets. 
+        default (str, optional): The default value of the parameter. 
+        type (Literal["DataField"]): A description of the parameter's type. 
+
+    Example:
+
+    .. code-block:: python
+
+        from input_parameters import DataField
+
+        datafield = DataField(
+                        name="variable_name",
+                        default="tas",
+                        facet="variable",
+                        max_items=1,
+                        group=2,
+                        predefined_facet={"time_frequency":["1hr"]},
+                        help="Select the variable name"
+        )
+    """
+    facet: str = Field()
+    group: int = 1
+    multiple: bool = False
+    predefined_facets: Optional[Dict[str, Union[str, List[str]]]] = None
+    default: str = None
+    type: Literal["DataField"] = "DataField"
+
+class Range(BaseParam):
+    """
+    A simple Range parameter for iterable, indexable sequences such as lists and tuples.
+
+    Args:
+        default (List[BaseParamType], optional): The default value of the parameter. Defaults to None.
+        type (Literal["Range"]): A description of the parameter's type. Must be "Range".
+
+    Attributes:
+        default (List[BaseParamType]): The default value of the parameter. 
+        type (Literal["Range"]): A description of the parameter's type. 
+    """
+    default: Optional[List[BaseParamType]] = None
+    type: Literal["Range"] = "Range"
+
+class File(BaseParam):
+    """
+    A simple parameter to define a file stored on the system.
+
+    Args:
+        default (Path, optional): The default value of the parameter. Defaults to None.
+        type (Literal["File"]): A description of the parameter's type. Must be "File".
+
+    Attributes:
+        default (Path): The default value of the parameter. 
+        type (Literal["File"]): A description of the parameter's type. 
+    """
+    default: Optional[Path] = None
+    type: Literal["File"] = "File"
+
+    
+    @field_validator("default")
+    @classmethod
+    def assert_is_file(cls, path: Path):
+        if path and not path.is_file():
+            raise ValueError("Default value must be a file!")
+        return path
+
```

Comment:
how about Solrfield? it;s the most important one i think

Reply:
I defined a field called `DataField` that takes over this role. I wanted to make it more generic, as the backend for the databrowser might not be Solr.